File: python/p_0007.py
#!/usr/bin/env python3

"""What is the 10 001st prime number?"""


def find_prime_number(count):
    """Find `count`th prime number."""
    prime_num = 0
    candidate = 1
    primes = list()
    while prime_num < count:
        candidate += 1
        is_prime = True
        # for i in range(2, int(candidate/2)+1):
        for i in primes:
            if candidate % i == 0:
                is_prime = False
                break
        if is_prime:
            primes.append(candidate)
            prime_num += 1
    print(count, "th prime:", candidate)
    # print(primes)


if __name__ == '__main__':
    print(find_prime_number(10001))

File: cpp/p_0007.cc
#include <algorithm>
#include <cmath>
#include <iostream>
#include <vector>

int find_prime_number(int count) {
    std::vector<int> primes;
    int candidate = 2;
    while (primes.size() < count) {
        ++candidate;
        if (std::none_of(primes.cbegin(), primes.cend(),
                         [candidate](int i) { return candidate % i == 0; })) {
            primes.push_back(candidate);
        }
    }
    return primes.back();
}

int main() {
    int nth_prime = find_prime_number(10001);
    std::cout << "Problem 0007: The 10001st prime number is: " << nth_prime
              << std::endl;
    return 0;
}
